llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -37262,6 +37262,27 @@ void X86TargetLowering::computeKnownBitsForTargetNode(const SDValue Op,
     }
     break;
   }
+  case X86ISD::HADD: {
+    Known = DAG.computeKnownBits(Op.getOperand(0), DemandedElts, Depth + 1);
```

Comment:
You can't use the result DemandedElts for the source operands you'll need something like:
```cpp
    APInt DemandedLHS, DemandedRHS;
    getHorizDemandedElts(VT, DemandedElts, DemandedLHS, DemandedRHS);
```
You can avoid computeKnownBits calls for cases where either DemandedLHS/RHS are zero.
Plus you can probably get more refined knownbits by getting the known bits of the odd/even elements separately, although that could mean 4 computeKnownBits calls.

Reply:
I've added the implementation with 4 `computeKnownBits` calls in the hope that it does not slow down the analysis too much.